### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds a configuration for the "typos" tool (`cargo install typos`) plus a github-actions job for it (that currently does not fail CI if it finds something). It also fixes all typos found by the tool.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
And we also got a showcase how my patch for disabling CI failures with this job did not work. I will fix this up of course.

**Comment by matthiasbeyer** [comment]
Reverted the fixup to check whether CI fails with typos now.

**Comment by matthiasbeyer** [comment]
Okay, works.

Going to clean this branch now.

File: configuration/contrib/collectd/collectd.conf [hunk 1]
```diff
@@ -512,7 +512,7 @@ LoadPlugin mqtt
 </Plugin>
 
 #<Plugin disk>
-#	Disk "hda"
+#	Disk "had"
```
Comment (didier-wenzek) [inline configuration/contrib/collectd/collectd.conf:515]: This needs to be reverted.

How can we avoid this kind of false negative? `default.extend-words` ? `extend-exclude` ? In a configuration file, as here, I prefer the later.

**Review by didier-wenzek — CHANGES_REQUESTED** [review]
I see this as really helpful - despite a risk of wrong suggestions as the one I have been lucky to spot. Hence, I have a slight concerns of having people blindly applying suggestions.

File: configuration/contrib/collectd/collectd.conf (hunk 1 above)
Comment (matthiasbeyer) [reply configuration/contrib/collectd/collectd.conf:515]: Good catch!

I will add an exception for "hda".

**Review by didier-wenzek — APPROVED** [review]
Let's give a try to this. I see a risk of typo fixes that break code or config. However, this risk is manageable.

**Comment by matthiasbeyer** [comment]
Cool! Rebased and squashed the fixup commit.

**Pull request merged**